services/meta/client.go

```diff
@@ -400,6 +400,37 @@ func (c *Client) User(name string) (*UserInfo, error) {
 	return nil, ErrUserNotFound
 }
 
+// UserWithPassword returns a user if the given password is correct.
+func (c *Client) UserWithPassword(username, password string) (*UserInfo, error) {
```

Comment:
It looks like there is a function called `Authenticate` which does the same thing as this. Will that work instead of adding a new function?